packages/components/src/autocomplete/autocompleter-ui.js

```diff
@@ -40,7 +48,16 @@ export function getAutoCompleterUI( autocompleter ) {
 			value,
 		} );
 
+		const [ needsA11yCompat, setNeedsA11yCompat ] = useState( false );
 		const popoverRef = useRef();
+		const popoverRefs = useMergeRefs( [
+			popoverRef,
+			useRefEffect( ( node ) => {
+				if ( node.ownerDocument !== contentRef.current.ownerDocument ) {
+					setNeedsA11yCompat( true );
+				}
+			} ),
```

Comment:
We could add a code comment here to give more context? Maybe something like

```suggestion
			useRefEffect( ( node ) => {
				// If the list of suggestions is rendered in another `Document`, enable
				// an "a11y compat" layer which duplicates the list contents in that
				// `Document`. This allows `aria-*` props to find the matching Element.
				if ( node.ownerDocument !== contentRef.current.ownerDocument ) {
					setNeedsA11yCompat( true );
				}
			} ),
```